pandas/tests/scalar/period/test_period.py

```diff
@@ -767,6 +768,40 @@ def test_period_deprecated_freq(self):
             assert isinstance(p1, Period)
             assert isinstance(p2, Period)
 
+    def period_constructor(bound, offset):
+        return Period(
+            year=bound.year,
+            month=bound.month,
+            day=bound.day,
+            hour=bound.hour,
+            minute=bound.minute,
+            second=bound.second + offset,
+            freq="us",
+        )
+
+    @pytest.mark.parametrize("bound, offset", [(Timestamp.min, -1), (Timestamp.max, 1)])
+    def test_start_time_and_end_time_bounds(self, bound, offset):
+        # GH #13346
+        period = TestPeriodProperties.period_constructor(bound, -offset)
+        assert period.start_time.round(freq="S") == period.to_timestamp().round(
```

Comment:
Could you parameterize over `start_time` and `end_time` so you can just call `getattr(period, time_variable).method(...)`

Reply:
The tests were splitted in two tests `test_outter_bounds_start_and_end_time` and `test_inner_bounds_start_and_end_time` with a better parametrization.
Thanks for the tips, I didn't knew about `getattr` function.